repos/system_upgrade/el7toel8/actors/cupsscanner/libraries/cupsscanner.py

```diff
@@ -93,21 +93,30 @@ def include_directive_check(read_func=_read_file):
     included_files = ['/etc/cups/cupsd.conf']
     error_list = []
 
-    for included_file in included_files:
+    vetted_included_files = []
+    while included_files:
+        # NOTE(ivasilev) Will be using stack to process last encountered include directives first
+        included_file = included_files.pop(-1)
```

Comment:
```suggestion
        for included_file in included_files.copy():
```
?

Reply:
That is not the only place, another violation is in remove part. I felt pretty bad we missed that in original pr so I decided to refactor the implementation completely (as the functionality is decently covered by tests I can be sure my version has the same behaviour). Too much in your opinion? You'd rather leave it be?

Reply:
I just realized you're right, I didn't get the whole context.